src/ophyd_async/epics/pvi/pvi.py

```diff
@@ -185,7 +192,7 @@ def _sim_common_blocks(device: Device, stripped_type: Optional[Type] = None):
             signal_type = args[0] if (args := get_args(sub_device_t)) else None
             sub_device = sub_device_t(SimSignalBackend(signal_type, sub_name))
         else:
-            sub_device = sub_device_t()
+            sub_device = getattr(device, sub_name, sub_device_t())
```

Comment:
What are the cases when this might or might not exist as an attribute?

Reply:
This just gets the object created in `pre_initialize_blocks`:

If the device was made in `pre_initialize_blocks` in `HdfPanda.__init__`, then we can't create a new object, since those references will have been given to devices that `HdfPanda` has initialized.